test/built-ins/TypedArrays/of/custom-ctor-returns-smaller-instance-throws.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es6id: 22.2.2.2
+esid: sec-%typedarray%.of
+description: >
+  Throws a TypeError if a custom `this` returns a smaller instance
+info: |
+  %TypedArray%.of ( ...items )
+
+  1. Let len be the actual number of arguments passed to this function.
+  ...
+  5. Let newObj be ? TypedArrayCreate(C, « len »).
+  ...
+includes: [testTypedArray.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var Other = TA === Int8Array ? Uint8Array : Int8Array;
+  var custom = new Other(1);
```

Comment:
We don't need the instance to be derived from a different constructor for this test, so it would be simpler to write `var custom = new TA(1);`